src/main/java/com/example/cinematicketbookingapp/repository/SeatRepository.java

```diff
@@ -0,0 +1,24 @@
+package com.example.cinematicketbookingapp.repository;
+
+import com.example.cinematicketbookingapp.model.Seat;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Lock;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.stereotype.Repository;
+
+import javax.persistence.LockModeType;
+import java.util.List;
+import java.util.Optional;
+
+@Repository
+public interface SeatRepository extends JpaRepository<Seat, Long> {
+
+    @Query("SELECT seat from Seat seat where seat.screeningRoom.id=" +
+            "(SELECT distinct screening.screeningRoom.id FROM Screening screening where screening.id=:screeningId)" +
```

Comment:
Można było zamiast podzapytania zrobić po prostu joina chyba?

Reply:
W zasadzie to `INNER JOIN` również by się nadał i z tego co czytałem na temat wydajności tych dwóch podobnych sposobów (`nested SELECT` oraz `INNER JOIN`) to `INNER JOIN` jest w większości przypadków wydajniejszy, natomiast przynajmniej dla mnie, podzapytania są czytelniejsze (a przynajmniej w tym przypadku). Tak wyglądałoby zapytanie w `JPQL` dla `INNER JOINA`: 
`@Query("SELECT seat from Seat seat INNER JOIN Screening screening` 
`ON seat.screeningRoom.id = screening.screeningRoom.id` 
`WHERE screening.id=:screeningId` 
`ORDER BY seat.rowNumber ASC, seat.seatNumber ASC")`

Reply:
Optymalizator i tak zrobi z tym co będzie chciał, przynajmniej przy tak prostych zapytaniach ;) Natomiast dobrze jest dostosować się do reszty zespołu - jeśli zespół wybierze JOIN to pewnie trzeba tak pisać.